tests/test_volatility_updates.cairo

```diff
@@ -0,0 +1,330 @@
+
+    %{
+        ids.half_size = int(ids.option_size / 2)
+        ids.option_size = ids.half_size * 2 # To prevent rounding errors I guess
+
+        if ids.option_type == 0: 
+            ids.pool_balance = ids._pool_balance
+        elif ids.option_type == 1: 
+            ids.pool_balance = int(ids._pool_balance * (ids.strike / 2**61))
+        else:
+            raise ValueError(f"Unknown option type: {ids.option_type}")
+   %}
+
+    //////////////////////////////////////
+    // Volatility update formula
+    //
+    // sigma_{trade} = (sigma_{t-1} + sigma_{t}) / 2
+    // 
+    // vol_denom = 1 - (Q_{t} / PS_{t}) ^ alpha
+    // sigma_{t} = sigma_{t-1} * 1 / vol_denom
+    //
+    // Note: alpha is 1
+    //////////////////////////////////////
+
+    // Basic update
+    let (desired_vol, _) = get_new_volatility(
+        current_volatility = volatility,
+        option_size = option_size,
+        option_type = option_type,
+        side = trade_side,
+        strike_price = strike,
+        current_pool_balance = pool_balance
+    );
+
+    %{
+        # Save the value as the reference gets revoked
+        context.desired_vol = ids.desired_vol
+    %}
+
+    // Same update, but divided into two parts  
+    let (vol_1, _) = get_new_volatility(
+        current_volatility = volatility,
+        option_size = half_size,
+        option_type = option_type,
+        side = trade_side,
+        strike_price = strike,
+        current_pool_balance = pool_balance
+    );
+
+    // Adjust pool balance 
+    local pool_balance_2;
+    %{  
+        if ids.trade_side == 0:
+            if ids.option_type == 0:
+                ids.pool_balance_2 = ids.pool_balance - ids.half_size
+            elif ids.option_type == 1:
+                ids.pool_balance_2 = ids.pool_balance - int(ids.half_size * (ids.strike / 2**61))
+        elif ids.trade_side == 1:
+            ids.pool_balance_2 = ids.pool_balance
+        else:
+            raise ValueError(f"Unknown trade side: {ids.trade_side}")
+    %}
+
+    let (vol_2, _) = get_new_volatility(
+        current_volatility = vol_1, // Use previous vol
+        option_size = half_size,
+        option_type = option_type,
+        side = trade_side,
+        strike_price = strike,
+        current_pool_balance = pool_balance_2
+    );  
+
+    %{
+        from math import isclose
+        error_string = f"""
+            Test get new vol failed for:
+            option_type = {ids.option_type}, 
+            side = {ids.trade_side} ,
+            pool_balance = {ids.pool_balance}, 
+            size = {ids.option_size}, 
+            init_vol = {ids.volatility}, 
+            desired_vol = {context.desired_vol},
+            final_vol = {ids.vol_2}
+        """
+
+        assert isclose(context.desired_vol, ids.vol_2, rel_tol = 0.01), error_string
+    %}
+
+    return();
+}
+
+@external
+func setup_volatility_updates{syscall_ptr: felt*, range_check_ptr}(){
+
+    deploy_setup();
```

Comment:
Not sure what the real reason is, but it's good to not run everything in t_r_integ_tests, they are the bottleneck on test runtime since protostar only runs different test suites in parallel

Reply:
but protostar doesn't paralelize the deploy_setup(), right? they paralelize the consequtive tests...

Reply:
They don't parallelize tests run from the same file, even though they could. So everything from test_run_integration_tests.cairo runs sequentially